src/Autofac/Core/Resolving/Pipeline/ResolveRequestContext.cs

```diff
@@ -23,40 +23,106 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 // OTHER DEALINGS IN THE SOFTWARE.
 
+using System;
+using System.Collections.Generic;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Autofac.Features.Decorators;
 
 namespace Autofac.Core.Resolving.Pipeline
 {
     /// <summary>
-    /// Context area for a resolve request.
+    ///     Context area for a resolve request.
     /// </summary>
-    internal sealed class ResolveRequestContext : ResolveRequestContextBase
+    internal class ResolveRequestContext : IResolveRequestContext
```

Comment:
Can this stay sealed if we're exposing an interface? Not sure why we'd need to derive from it now.

Reply:
`PipelineBuilderTests`  class contains a `MockPiplineRequest` which derived from `ResolveRequestContextBase`. That's why I've unsealed it. Should we change that?

Reply:
You could argue that it should now just implement the interface? 

This feeds into the argument for revisiting the pipeline tests we have to use the new approach to mocking the context.

Reply:
Agree, would like to do that in a separate PR if you don't mind.